src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -250,50 +246,34 @@ protected override void RequestHeartbeat()
             }
         }
 
-        private async Task MonitorServersAsync()
+        private void ServerDescriptionChangedHandler(object sender, ServerDescriptionChangedEventArgs args)
```

Comment:
This method is called (via event handlers) from async code and therefore should not block (or block for long).

I think we're fine, and even if we do block it shouldn't be often enough or long enough to gum up the async machinery.

Reply:
Noted!